streams/src/main/java/org/apache/kafka/streams/processor/ProcessorRecordContext.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor;
+
+/**
+ * An {@link RecordContext} that also can forward to children of
+ * the current {@link org.apache.kafka.streams.processor.internals.ProcessorNode}
+ */
+public interface ProcessorRecordContext extends RecordContext {
```

Comment:
Yeah I understand that. What I had in mind is that in order to not change the public APIs, we copy-paste the fields of topic/partition/offset/timestamp in `RecordContext` to `ProcessorContextImpl`, and when flushing is triggered we overwrite these fields with the values we read from the deserialized `RecordContext`, then call `ProcessorContextImpl.forward`. Reading your code again I realize this may not be too much different compared with introducing a new wrapper class though.